src/lib.rs

```diff
@@ -10,15 +10,59 @@ use wasm_bindgen::JsCast;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
-#[wasm_bindgen]
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum Color {
+    Red,
+    Pink,
+    Blue,
+    Orange,
+    White,
+    Black,
+}
+
+impl Color {
+    fn as_js_value(&self) -> JsValue {
+        JsValue::from_str(match self {
+            Color::Red => "red",
+            Color::Pink => "hotpink",
+            Color::Blue => "blue",
+            Color::Orange => "orange",
+            Color::White => "white",
+            Color::Black => "black",
+        })
+    }
+
+    fn random() -> Color {
+        let rand = js_sys::Math::random();
+        match (rand * 6.0).floor() as u32 {
+            0 => Color::Red,
+            1 => Color::Pink,
+            2 => Color::Blue,
+            3 => Color::Orange,
+            4 => Color::White,
+            _ => Color::Black,
+        }
+    }
+}
+
+// impl rand::distributions::Distribution<Color> {}
+
 #[derive(Clone)]
-pub struct Game {
+pub struct Player {
+    color: Color,
     x: f64,
     y: f64,
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
```

Comment:
porque?

Reply:
We clone our Game on line 234 to make the borrow checker happy. `.clone()` makes a totally separate deep copy of a value: https://doc.rust-lang.org/std/clone/trait.Clone.html